### ISSUE TITLE
Save original join date in addition to instance join date

### ISSUE THREAD

**Issue by barts** [mastodon/mastodon#20412]
### Pitch

I joined Mastodon in 2018 and noticed that if I attempted to move instances, the “Joined” date became a 2022 date. I wish in that case that the new instance was able to migrate over my original join date. Or if that’s already accessible in some kind of account migration history? In either case it would be nice if my profile displayed something like “Instance joined Nov 11, 2022 (Joined Mastodon Mar 26, 2018)”.

### Motivation

The fewer reasons people have for not wanting to move instances, the better. For Mastodon as a whole. You are already putting some importance to the join date by displaying it at the top of people’s profiles. And I’ve seen people say things like “hey it looks like I’ve just joined but actually I’ve been a member since #### and just moved instances”.

**Comment by Danie10** [comment]
Yes would be good to retain that original join date if possible. I actually was specifically searching about this now, which is why I found this issue logged.

**Comment by yawnbox** [comment]
Is there a straight forward DB edit process for doing this as an admin for users?

**Comment by frankramblings** [comment]
> Is there a straight forward DB edit process for doing this as an admin for users?

I have the same question!

**Comment by frankramblings** [comment]
@yawnbox This worked a treat! Thanks very much. 😄

**Comment by MeesJ** [comment]
This is definitely one of the things that should be added. It makes much more sense to make the join date be the date an account joined Mastodon rather than the instance it's on.

**Comment by petrarca-arezzo** [comment]
I'm confused with this "Join-Date" shown in the profile as well - it's listed very prominently at the top so must be considered one the most important information about someone.

I have the suspicion this date is simple the date when the specific server-account was first created, which I think is rather useless information as people might move around between different servers at any time (and always get a new join-date).

I would prefer either
a) show the date when joined Mastodon (whatever server); so the date remains the same also when moving accounts to other servers
b) simply fully forget about this date, remove it, don't show it anymore anywhere.

**Comment by erebion** [comment]
`b)` yes, please

(`a)` does not work when the first instance is no longer a available. I used several instances which are defunct now, so I cannot even look up the precise date I joined.)

**Comment by frankramblings** [comment]
> (`a)` does not work when the first instance is no longer a available. I used several instances which are defunct now, so I cannot even look up the precise date I joined.)

This is a fair point. I wonder — in a situation like yours, would it be possible to default to the oldest-available join date? For example, if you’ve used 3 different instances and the first is defunct, could it default to showing the join date for the second one?

If not, then I agree it probably makes the most sense to drop Join Date from the profile or make it optional.

**Comment by barts** [comment]
Ideally the original join date would travel with you when you’re migrating to a new instance, along with any other metadata that gets transferred over. So you wouldn’t need the original instance to be up and running, the original join date would be fetched from your profile in your current instance.

It would be nice if it was able to preserve both your original join date and the date you joined your current instance and present those nicely if the two are different.

To some communities the instance join date might be important if they’re topic-focused or require approval to join. Then it’s useful to see how new someone is to that particular instance’s rules/code of conduct/etc.

**Comment by erebion** [comment]
>This is a fair point. I wonder — in a situation like yours, would it be possible to default to the oldest-available join date? For example, if you’ve used 3 different instances and the first is defunct, could it default to showing the join date for the second one?

Either allow me as a user to just set it myself or remove it entirely.

**Comment by erebion** [comment]
At least don't force people to have a date displayed that does not convey any actually useful info. I've even switched accounts on the aame instance so that it does not even say when I joined the instance. So people assumed i was new and gave me tips on how to interact while I had used it for 3 years already...

**Comment by erebion** [comment]
And then there's also privacy, I just might not want to tell anyone. I wonder how it ever came to be a feature as I cannot see ita use.
